postcommit.py

```diff
@@ -0,0 +1,642 @@
+import mercurial, urllib, subprocess
+from mercurial import util
+from mercurial import patch
+from mercurial.ui import ui
+import json, os, inspect, sys, urllib2, httplib, datetime, re
+
+from mercurial.localrepo import localrepository
+
+
+repopath = sys.argv[1]
+type = sys.argv[2]
+if len(sys.argv) > 4:
+	oldId = sys.argv[3]
+	newId = sys.argv[4]
+	branch = sys.argv[5]
+elif len(sys.argv) > 3:
+	st_changeset = sys.argv[3]
+else:
+	st_changeset = ""
+
+
+COMMIT_TEMPLATE = 'LEGAL DECLARATION:I hereby ([^\n])*\n(LICENSE COMPATIBLE:([Yy][Ee][Ss]|[Nn][Oo]))*\n*'
+OPTIONAL_TEMPLATE = 'IGNORE_ISSUECHECKS:YES(\n)*(I18N ISSUE ID :)*'
+
+def getFileType(filename):
+    values = filename.split('.')
+    if len(values) > 1:
+        return values[-1]
+    else:
+        return 'others'
+
+def escapeChars(string):
+    string = string.strip('\n').strip().strip('\\').strip('/')
+    string = string.replace("\\", "\\\\").replace("'","\\'").replace("+", "\\+").replace("$", "\$")
+    return string
+
+def striplicense_template(logmsg):
+
+    description = re.sub(COMMIT_TEMPLATE,'',logmsg)
+    description = re.sub(OPTIONAL_TEMPLATE,'',description)
+    description = escapeChars(description)
+    return description
+
+def addLogs(stringVal):
+    print stringVal
+
+def getCurrentTime():
+    now = datetime.datetime.now()
+    cur_time = now.strftime('%Y-%m-%d %H:%M:%S')
+    return cur_time
+
+def status_check(exec_command, exec_dir=None, background=False):
+
+    if exec_dir:
+        os.chdir(exec_dir)
+
+    res = subprocess.Popen(exec_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if not background:
+        result = res.communicate()
+        return result
+
+class HGRepository(object):
+
+    def __init__(self, repo, ui):
+
+        self.repo_type = 'HG'
+        self.ui = ui
+        self.repo = repo
+        self.getStartChangeset()
+        self.index_node = self.repo[self.node].parents()[0].hex()[0:11]
+        self.module = os.path.basename(repo.root)
+        self.branch = self.repo['tip'].branch()
+        self.st_changeset = str(self.repo[self.node].rev()) + ":" + str(self.node)
+        self.end_changeset = str(self.repo['tip'].rev()) + ":" + str(self.repo['tip'])
+        self.commit_details = {}
+        self.branch_details = {}
+        self.diff_needed_files = {}
+        self.diff_content = {}
+
+    def getStartChangeset(self):
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "getStartChangeset" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)